tests/test_flows.py

```diff
@@ -132,6 +133,47 @@ def __capture(self, test_index, test_name, bridge, duration=None):
                     tshark_args+['-a', 'duration:900'], shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         except subprocess.CalledProcessError as e:
             self.err(f'test {self.run_tests} failed, output: {e.output}')
+    def check_tshark(self, target_thread_name=None, end=time.time(), start=None, file_output=False):
+        '''
+        ### Description
+        Looks into the tshark captured data, updating the global capture data
+        and returning the captured data from `start` to `end`
+
+        ### Parameters
+        `target_thread_name=None` (str) dictionary key (in self.tshark_instances) of the thread running tshark, if None, the method
+        defaults to self.running (i.e. the current running test)
+
+        `end=time.time()` (datetime) upper bound for captured packets timestamp
+
+        `start=None` (datetime) lower bound for captured packets timestamp
+
+        `file_output=False` flag to determine if the captured data would be 
+        written to a file
+
+        ### Returns
+
+        a collection of captured `ieee1905` packets
+        '''
+        # if file_output:
+        #     with open(self.outputfile, "w+") as f:
+        #         f.write(f'\n{time.time()} - test {self.running}\n')
+        if not target_thread_name:
```

Comment:
Actually we can avoid using if altogether:
`tshark_proc = self.tshark_instances[target_thread_name or self.running].proc`
it will take `target_thread_name` if it's not None, and `self.running` otherwise.

Reply:
Python is cool